packages/table/src/TableContext.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { createContext, useContext, useMemo, useReducer } from 'react';
+
+const TableTypes = {
+  RegisterColumn: 'REGISTER_COLUMN_INFO',
+  SortTableData: 'SORT_TABLE_DATA',
+  SetHasNestedRows: 'SET_HAS_NESTED_ROWS',
+  SetHasRowSpan: 'SET_HAS_ROW_SPAN',
+} as const;
+
+type TableTypes = typeof TableTypes[keyof typeof TableTypes];
+
+export { TableTypes };
+
+interface ActionPayload<T = {}> {
+  [TableTypes.RegisterColumn]: {
+    dataType?: DataType;
+    index: number;
+  };
+  [TableTypes.SetHasRowSpan]: boolean;
+  [TableTypes.SetHasNestedRows]: boolean;
+  [TableTypes.SortTableData]: {
+    columnId: number;
+    accessorValue: (data: any) => string;
+    data: Array<T>;
```

Comment:
Why do we need to include data in here? We can pull that straight from state, no need to pass it through an action.

Reply:
👍